File: src/main/java/com/todolist/task/service/domain/TaskService.java
package com.todolist.task.service.domain;

import com.todolist.task.service.domain.models.Task;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class TaskService {
  private final TaskRepository taskRepository;
  public void save(Task task) {
    taskRepository.save(task);
  }

  public List<Task> findByCreatedByAndDeletedFalse(String user) {
    return taskRepository.findByCreatedByAndDeletedFalse(user);
  }

  public Task findById(String taskId) {
    return taskRepository.findById(taskId);
  }
}

File: src/main/kotlin/com/todolist/task/service/domain/TaskService.kt
package com.todolist.task.service.domain

import com.todolist.task.service.domain.models.Task
import org.springframework.stereotype.Service

@Service
class TaskService(
    private val taskRepository: TaskRepository
) {
    fun save(task: Task) {
        taskRepository.save(task)
    }

    fun findByCreatedByAndDeletedFalse(user: String): List<Task>? {
        return taskRepository.findByCreatedByAndDeletedFalse(user)
    }

    fun findById(taskId: String): Task? {
        return taskRepository.findById(taskId)
    }
}
